packages/dashboard/src/components/Section/examples/EditSections.Customization.Example.tsx

```diff
@@ -0,0 +1,280 @@
+  };
+
+  /**
+   * Example handler for onDeleteSection
+   */
+  private _onDeleteSection = (id: string): void => {
+    const newSections = this.state.sections.filter((section: ISection) => section.id !== id);
+    this.setState({
+      ...this.state,
+      saveButtonDisabled: false,
+      sections: newSections
+    });
+  };
+
+  /**
+   * Example handler for onRenameSectionClick
+   */
+  private _onRenameSectionClick = (id: string): void => {
+    const index = this.state.sections.findIndex((section: ISection) => section.id === id);
+    const newSections = JSON.parse(JSON.stringify(this.state.sections)); // deep clone
+    newSections[index].isRenaming = true;
+
+    this.setState({
+      ...this.state,
+      sections: newSections
+    });
+  };
+
+  /**
+   * Example handler for onUpdateSectionTitle
+   */
+  private _onUpdateSectionTitle = (id: string, title: string) => {
+    const index = this.state.sections.findIndex((section: ISection) => section.id === id);
+    const newSections = JSON.parse(JSON.stringify(this.state.sections)); // deep clone
+    newSections[index].isRenaming = false;
+    newSections[index].title = title;
+
+    this.setState({
+      ...this.state,
+      saveButtonDisabled: false,
+      sections: newSections
+    });
+  };
+
+  /**
+   * Example handler for onCancel, which reload the sections and layout with the orignal value
+   */
+  private _onCancel = () => {
+    this.setState({
+      ...this.state,
+      saveButtonDisabled: true,
+      sections: this._sections(),
+      layout: this._generateLayout()
+    });
+  };
+
+  private _addLayoutForNewSection(id: string): DashboardGridBreakpointLayouts {
+    const newLayout: DashboardGridBreakpointLayouts = JSON.parse(JSON.stringify(this.state.layout));
+
+    for (const [_, value] of Object.entries(newLayout)) {
+      if (value === undefined) {
+        continue;
+      }
+      value.push({
+        i: id,
+        y: this._getMaxY(value) + CardSizeToWidthHeight[CardSize.section].h, // the new section should be placed at the very bottom
+        x: 0,
+        size: CardSize.section
+      });
+    }
+
+    return newLayout;
+  }
+
+  private _generateLayout(): DashboardGridBreakpointLayouts {
+    // this is an example. If layout is stored in the storage, the layout could be fetched from storage
+    return {
+      lg: [
+        { i: '0', y: 0, x: 0, size: CardSize.section },
+        { i: '1', y: 1, x: 0, size: CardSize.section },
+        { i: '2', y: 2, x: 0, size: CardSize.section }
+      ]
+    };
+  }
+
+  private _getMaxY(value: IDashboardCardLayout[]): number {
+    let maxY = 0;
+    for (let i = 0; i < value.length; i++) {
+      if (value[i].y > maxY) {
+        maxY = value[i].y;
+      }
+    }
+
+    return maxY;
+  }
+
+  private _generateRandomNewSectionId(): string {
+    // this is an example to generate an random id. Could be improved by using a guid
+    return 'newSection' + Math.floor(Math.random() * 101).toString();
```

Comment:
If it's to avoid conflict, my guess is we can still have a conflict with random sections